supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala

```diff
@@ -0,0 +1,629 @@
+    val evaluatedAttributes = TryUtil.sequenceMap(evaluateAttrs, "Runtime attributes evaluation").get
+    JesRuntimeAttributes(evaluatedAttributes)
+  }
+  private lazy val workingDisk: JesAttachedDisk = runtimeAttributes.disks.find(_.name == JesWorkingDisk.Name).get
+  private lazy val gcsExecPath: Path = callRootPath.resolve(JesExecScript)
+  private lazy val cmdInput = JesFileInput(ExecParamName, gcsExecPath.toString, Paths.get(JesExecScript), workingDisk)
+  private lazy val jesCommandLine = s"/bin/bash ${cmdInput.containerPath.toAbsolutePath.toString}"
+  private lazy val defaultMonitoringOutputPath = callRootPath.resolve(JesMonitoringLogFile)
+  private lazy val returnCodeFilename = s"${jesLogBasename(jobDescriptor.key)}-rc.txt"
+  private lazy val returnCodeGcsPath = callRootPath.resolve(returnCodeFilename)
+  private lazy val rcJesOutput = JesFileOutput(returnCodeFilename, returnCodeGcsPath.toString, Paths.get(returnCodeFilename), workingDisk)
+  private lazy val jesStdoutFile = callRootPath.resolve(s"${jesLogBasename(jobDescriptor.key)}-stdout.log")
+  private lazy val jesStderrFile = callRootPath.resolve(s"${jesLogBasename(jobDescriptor.key)}-stderr.log")
+  private lazy val standardParameters = Seq(rcJesOutput)
+  private lazy val jesLogFilename = s"${jesLogBasename(jobDescriptor.key)}.log"
+  // PBE this shouldn't need to be defined here, could be at the backend level.
+  private lazy val jesAttributes = JesAttributes(backendConfig)
+  private lazy val returnCodeContents = Try(returnCodeGcsPath.toAbsolutePath.contentAsString)
+  private lazy val dockerConfiguration: Option[JesDockerCredentials] = DockerConfiguration.build(backendConfig).dockerCredentials map JesDockerCredentials.apply
+  private lazy val maxPreemption = runtimeAttributes.preemptible
+  private lazy val preemptible: Boolean = jobDescriptor.key.attempt <= maxPreemption
+  private lazy val tag = self.path.name
+
+  private def globOutputPath(glob: String) = callRootPath.resolve(s"glob-${glob.md5Sum}/")
+
+  private def gcsAuthFilePath: Path =  {
+    // If we are going to upload an auth file we need a valid GCS path passed via workflow options.
+    val bucket = workflowDescriptor.workflowOptions.get(AuthFilePathOptionKey) getOrElse workflowPath.toString
+    bucket.toPath(gcsFilesystem).resolve(s"${workflowDescriptor.id}_auth.json")
+  }
+
+  private def authGcsCredentialsPath(gcsPath: String): JesInput = JesLiteralInput(ExtraConfigParamName, gcsPath)
+
+  private def gcsAuthParameter: Option[JesInput] = {
+    if (jesAttributes.gcsFilesystemAuth.requiresAuthFile || dockerConfiguration.isDefined)
+      Option(authGcsCredentialsPath(gcsAuthFilePath.toString))
+    else None
+  }
+
+  private val callEngineFunctions = {
+    val callContext = new CallContext(
+      callRootPath.toString,
+      jesStdoutFile.toString,
+      jesStderrFile.toString
+    )
+
+    new JesCallEngineFunctions(List(gcsFilesystem), callContext)
+  }
```

Comment:
Once https://github.com/broadinstitute/cromwell/pull/777 is merged, the engine functions will need to be made available through the BackendFactory (see https://github.com/broadinstitute/cromwell/pull/777/files#diff-2bf1773c6f31c2d645fb58eeb920bb7fR13), so that the engine can evaluate the inputs for a Call.
In order to avoid duplicating code as much as possible, all the path making logics (or at least what's necessary to create a `CallContext`) would be better off on its own so it can be used both in the factory and in the BackendExecution.